assets/js/modules/tagmanager/datastore/accounts.js

```diff
@@ -0,0 +1,190 @@
+import { isValidAccountSelection } from '../util/validation';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+const { createRegistrySelector } = Data;
+
+// Actions
+const RESET_ACCOUNTS = 'RESET_ACCOUNTS';
+
+const fetchGetAccountsStore = createFetchStore( {
+	baseName: 'getAccounts',
+	controlCallback: () => API.get( 'modules', 'tagmanager', 'accounts', null, { useCache: false } ),
+	reducerCallback: ( state, accounts ) => {
+		return {
+			...state,
+			accounts: [ ...accounts ],
+		};
+	},
+} );
+
+export const BASE_INITIAL_STATE = {
+	accounts: undefined,
+};
+
+export const baseActions = {
+	*resetAccounts() {
+		const { dispatch } = yield Data.commonActions.getRegistry();
+
+		yield {
+			payload: {},
+			type: RESET_ACCOUNTS,
+		};
+
+		dispatch( STORE_NAME ).invalidateResolutionForStoreSelector( 'getAccounts' );
+	},
+
+	*selectAccount( accountID ) {
+		invariant( isValidAccountSelection( accountID ), 'A valid accountID is required to select.' );
+
+		const registry = yield Data.commonActions.getRegistry();
+		registry.dispatch( STORE_NAME ).setAccountID( accountID );
+
+		if ( ACCOUNT_CREATE === accountID ) {
+
+		}
+
+		// Trigger cascading selections.
+	},
+};
+
+export const baseReducer = ( state, { type } ) => {
+	switch ( type ) {
+		case RESET_ACCOUNTS: {
+			return {
+				...state,
+				accounts: undefined,
+				settings: {
+					...state.settings,
+					accountID: undefined,
+					ampContainerID: undefined,
+					containerID: undefined,
+					internalAMPContainerID: undefined,
+					internalContainerID: undefined,
+				},
+			};
+		}
+
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const baseResolvers = {
+	*getAccounts() {
+		const registry = yield Data.commonActions.getRegistry();
+		const existingAccounts = registry.select( STORE_NAME ).getAccounts();
+
+		// Only fetch accounts if there are none in the store.
+		if ( ! existingAccounts ) {
+			yield fetchGetAccountsStore.actions.fetchGetAccounts();
+		}
+	},
+};
+
+export const baseSelectors = {
+	/**
+	 * Gets all Google Tag Manager accounts this user can access.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {?Array.<Object>} An array of account objects; `undefined` if not loaded.
+	 */
+	getAccounts( state ) {
+		const { accounts } = state;
+
+		return accounts;
+	},
+
+	/**
+	 * Gets an error encountered by this store or its side effects.
+	 *
+	 * Returns an object with the shape when there is an error:
+	 * ```
+	 * {
+	 *   code,
+	 *   message,
+	 * }
+	 * ```
+	 *
+	 * Returns `null` if there was no error.
+	 *
+	 * Marked as private, because in the future we'll have more robust error
+	 * handling.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {?Object} Any error encountered with requests in state.
+	 */
+	getError( state ) {
+		const { error } = state;
+
+		return error || null;
+	},
```

Comment:
Done! Maybe we should add something like this to `commonStore` or `createModuleStore` instead? I took the same file from AdSense for consistency and only had to update doc blocks.

Reply:
I don't think we should do that for now, since we're still not clear whether this is the error handling we'll wanna have in the long term. While I agree this is bad for DRY, let's not abstract it for this reason just yet.